### ISSUE TITLE
reportPropertyTypeMismatch has a default of "none" but still produces a warning

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#2980]
**Describe the bug**
According to the docs, the default for the configuration parameter `reportPropertyTypeMismatch` is `"none"` which means that it should not be reported.  But it is still showing up as a warning.

**To Reproduce**
See code below

**Expected behavior**
No warning

**Screenshots or Code**
```python
from typing import Union


class Foo3:
    def __init__(self):
        self._x: int = 0

    @property
    def x(self) -> int:
        return self._x

    @x.setter
    def x(self, x: Union[int, str]) -> None:
        if isinstance(x, str):
            self._x = int(x)
        else:
            self._x = x

```

reports:
![image](https://user-images.githubusercontent.com/15113894/152379916-8bde8b8b-b747-457e-a5ee-053047a768df.png)

**VS Code extension or command-line**
VSCode 2022.1.5

**Additional context**

Release notes for VSCode 2021.12.0 said this was fixed...

**Comment by erictraut** [comment]
I'm not able to repro, and I've double-checked the code and confirmed that this setting is off by default. You must have it enabled in your settings, a pyrightconfig.json file, a pyproject.toml file, or a `# pyright: reportPropertyTypeMismatch` comment.

**Comment by Dr-Irv** [comment]
Yes, sorry, it was in my settings.
